src/iceberg/table_requirements.cc

```diff
@@ -134,4 +136,24 @@ Result<std::vector<std::unique_ptr<TableRequirement>>> TableRequirements::ForUpd
   return context.Build();
 }
 
+Result<bool> TableRequirements::IsCreate(
+    const std::vector<std::unique_ptr<TableRequirement>>& requirements) {
+  bool is_create = std::ranges::any_of(requirements, [](const auto& req) {
+    return dynamic_cast<table::AssertDoesNotExist*>(req.get()) != nullptr;
```

Comment:
Please add an enum type to `TableRequirement` like `TableUpdate::kind()` and use it here.